cpp/src/arrow/array/data.cc

```diff
@@ -130,6 +130,33 @@ int64_t ArrayData::GetNullCount() const {
   return precomputed;
 }
 
+bool ArrayData::Equals(const ArrayData& other) const {
```

Comment:
I don't think this is right. Equality is more complicated than that.
Instead of adding this, I would suggest you simply use `MakeArray` in tests and then call `AssertArraysEqual`.

Reply:
Thank you for your comment. I realized that this method is not used now. It was implemented for my old versions. Actually, what I committed is your suggestion :)
I will drop it.